File: cpp/lab2/source/plus.cpp
#include "operations.h"
#include "factory.h"

class Plus : public Operations{
public:
    void Operation(std::stack<int> &stack, std::istream& fin) override{
        Empty_check(stack);
        int previous_top = stack.top();
        stack.pop();
        Empty_check(stack);
        int res = previous_top + stack.top();
        stack.pop();
        stack.push(res);
    }

};

namespace {
    Operations* createPlus() {
        return new Plus;
    }
    bool x = OperationsFactory::getInstance()->RegisterOperation(std::string("+"), createPlus);
}

File: jaba/lab1/src/main/java/org/example/Plus.java
package org.example;

import java.io.OutputStream;
import java.util.Stack;
import java.util.Scanner;

public class Plus extends Operations{
    @Override
    public void Operation(Stack<Integer> stack, Scanner scanner, OutputStream s){
        emptyCheck(stack);
        int previous_top = stack.pop();
        emptyCheck(stack);
        int res = previous_top + stack.pop();
        stack.push(res);
    }
}
